iceoryx_hoofs/test/moduletests/error_reporting/test_helper.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <csetjmp>
+#ifndef IOX_HOOFS_MODULETESTS_ERROR_REPORTING_TEST_HELPER_HPP
+#define IOX_HOOFS_MODULETESTS_ERROR_REPORTING_TEST_HELPER_HPP
+
+#include "iceoryx_hoofs/cxx/function_ref.hpp"
+#include "iceoryx_hoofs/error_reporting/platform/default/error_handler.hpp"
+
+#include <thread>
+#include <utility>
+
+#include <functional>
+#include <stdio.h>
+
+#define ASSERT_NO_PANIC()                                                                                              \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        ASSERT_FALSE(iox::err::TestErrorHandler::instance().hasPanicked());                                            \
+    } while (false)
+
+#define ASSERT_PANIC()                                                                                                 \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        ASSERT_TRUE(iox::err::TestErrorHandler::instance().hasPanicked());                                             \
+    } while (false)
+
+#define ASSERT_ERROR(code)                                                                                             \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        auto e = error_code_t(code);                                                                                   \
+        ASSERT_TRUE(iox::err::TestErrorHandler::instance().hasError(e));                                               \
+    } while (false)
+
+namespace iox
+{
+namespace testing
+{
+
+/// @brief runs testFunction in a testContext that can detect fatal failures;
+/// runs in the same thread
+/// @note uses a longjump
+template <typename Function, typename... Args>
+void testContext(Function&& testFunction, Args&&... args)
+{
+    auto& buf = iox::err::TestErrorHandler::instance().prepareJump();
+
+    // setjmp must be called in a stackframe that still exists when longjmp is called
+    // Therefore there cannot be a convenient abstraction that does not also
+    // know the test function that is being called.
+    // NOLINTNEXTLINE
+    if (setjmp(&buf[0]) != iox::err::TestErrorHandler::instance().jumpIndicator())
+    {
+        testFunction(std::forward<Args>(args)...);
+    }
+}
```

Comment:
Yes. We talked about this during some coffee break. But this cannot be fixed IMO due to limitations of `setjmp`. `setjmp` and `longjmp` must happen in the same thread. I see no way to enforce this except stating limitations.

Reply:
It is a long time since I reviewed this but the easiest solution is to not have the jmp buf as part of the error handler but just as local variable. Some things would require a refactoring though.